src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/components/step_data_source/components/header/header.tsx

```diff
@@ -3,49 +3,88 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React, { useState } from 'react';
+import React, { Fragment, useState } from 'react';
 import './header.scss';
 
 import {
-  EuiTitle,
-  EuiSpacer,
-  EuiText,
-  EuiFlexItem,
   EuiButton,
   EuiFlexGroup,
+  EuiFlexItem,
   EuiRadio,
+  EuiSelectable,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
 } from '@elastic/eui';
 
-import { FormattedMessage } from '@osd/i18n/react';
 import { i18n } from '@osd/i18n';
+import { FormattedMessage } from '@osd/i18n/react';
+import { useEffectOnce } from 'react-use';
 import {
   DataSourceRef,
   IndexPatternManagmentContext,
 } from 'src/plugins/index_pattern_management/public/types';
-import { SavedObjectFinderUi } from '../../../../../../../../../plugins/saved_objects/public';
 import { useOpenSearchDashboards } from '../../../../../../../../../plugins/opensearch_dashboards_react/public';
-import { StepInfo } from '../../../../types';
+import { getDataSources } from '../../../../../../components/utils';
+import { DataSourceTableItem, StepInfo } from '../../../../types';
+import { LoadingState } from '../../../loading_state';
 
 interface HeaderProps {
-  onDataSourceSelected: (id: string, type: string) => void;
+  onDataSourceSelected: (id: string, type: string, title: string) => void;
   dataSourceRef: DataSourceRef;
   goToNextStep: (dataSourceRef: DataSourceRef) => void;
   isNextStepDisabled: boolean;
   stepInfo: StepInfo;
 }
 
-const DATA_SOURCE_PAGE_SIZE = 5;
-
 export const Header: React.FC<HeaderProps> = (props: HeaderProps) => {
   const { dataSourceRef, onDataSourceSelected, goToNextStep, isNextStepDisabled, stepInfo } = props;
   const { currentStepNumber, totalStepNumber } = stepInfo;
 
   const [defaultChecked, setDefaultChecked] = useState(true);
   const [dataSourceChecked, setDataSourceChecked] = useState(false);
+  const [dataSources, setDataSources] = useState<DataSourceTableItem[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const {
+    savedObjects,
+    notifications: { toasts },
+  } = useOpenSearchDashboards<IndexPatternManagmentContext>().services;
+
+  useEffectOnce(() => {
+    fetchDataSources();
+  });
+
+  const fetchDataSources = () => {
+    setIsLoading(true);
+    getDataSources(savedObjects.client)
+      .then((fetchedDataSources: DataSourceTableItem[]) => {
+        setIsLoading(false);
+        if (fetchedDataSources?.length) {
+          setDataSources(fetchedDataSources);
+        }
```

Comment:
Is it guarantied that the backend will throw if no data sources are available? If not, shouldn't we show something when no data sources are returned?

Reply:
I don't think backend will throw. Current behavior is the drop down will show as empty. in later phases, UX designed an empty state to show.